src/dmd/typinf.d

```diff
@@ -24,14 +24,26 @@ import dmd.mtype;
 import dmd.visitor;
 
 /****************************************************
- * Get the exact TypeInfo.
+ * Generates the `TypeInfo` object associated with `torig` if it
+ * hasn't already been generated
+ * Params:
+ *      loc   = the location for reporting line numbers in errors
+ *      torig = the type to generate the `TypeInfo` object for
+ *      sc    = the scope
  */
-extern (C++) void genTypeInfo(Type torig, Scope* sc)
+extern (C++) void genTypeInfo(Loc loc, Type torig, Scope* sc)
 {
     //printf("Type::genTypeInfo() %p, %s\n", this, toChars());
+
+    if (!global.params.useTypeInfo)
+    {
+        torig.error(loc, "`TypeInfo` cannot be used with -betterC");
+        fatal();
+    }
+
     if (!Type.dtypeinfo)
     {
-        torig.error(Loc.initial, "TypeInfo not found. object.d may be incorrectly installed or corrupt, compile with -v switch");
+        torig.error(loc, "`object.TypeInfo` was not found");
```

Comment:
Maybe add "but implicitly used" to the error message.

Reply:
Done.